site/app/controllers/AuthenticationController.php

```diff
@@ -308,4 +310,274 @@ public function userSelection() {
 
         return new WebResponse(AuthenticationView::class, 'userSelection', $users);
     }
+
+    /**
+     * Check if password has at least one of the following, Upper case letter, Lower case letter, Special character, and number
+     */
+    public function checkChars(string $password): bool {
+        $upperCase = preg_match('/[A-Z]/', $password);
+        $lowerCase = preg_match('/[a-z]/', $password);
+        $specialChar = preg_match('/[^A-Za-z0-9]/', $password);
+        $numericVal = preg_match('/[0-9]/', $password);
+        return $upperCase >= 1 && $lowerCase >= 1 && $specialChar >= 1 && $numericVal >= 1;
+    }
```

Comment:
It would be great to add some PHPUnit tests for this.  Also, what do you think about moving this to a utils library somewhere instead?  The function name could also be more descriptive.  The same comments apply to the following helper functions as well.